homeassistant/components/efergy/sensor.py

```diff
@@ -96,68 +136,55 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
-
-    def update(self):
+        if description.key == CONF_COST:
+            self._attr_native_unit_of_measurement = f"{currency}/{period}"
+
+    def update(self) -> None:
         """Get the Efergy monitor data from the web service."""
         try:
-            if self.type == "instant_readings":
+            if self.entity_description.key == CONF_INSTANT:
                 url_string = f"{_RESOURCE}getInstant?token={self.app_token}"
                 response = requests.get(url_string, timeout=10)
-                self._state = response.json()["reading"]
-            elif self.type == "amount":
+                self._attr_native_value = response.json()["reading"]
+            elif self.entity_description.key == CONF_AMOUNT:
                 url_string = f"{_RESOURCE}getEnergy?token={self.app_token}&offset={self.utc_offset}&period={self.period}"
                 response = requests.get(url_string, timeout=10)
-                self._state = response.json()["sum"]
-            elif self.type == "budget":
+                self._attr_native_value = response.json()["sum"]
+            elif self.entity_description.key == CONF_BUDGET:
                 url_string = f"{_RESOURCE}getBudget?token={self.app_token}"
                 response = requests.get(url_string, timeout=10)
-                self._state = response.json()["status"]
-            elif self.type == "cost":
+                self._attr_native_value = response.json()["status"]
+            elif self.entity_description.key == CONF_COST:
                 url_string = f"{_RESOURCE}getCost?token={self.app_token}&offset={self.utc_offset}&period={self.period}"
                 response = requests.get(url_string, timeout=10)
-                self._state = response.json()["sum"]
-            elif self.type == "current_values":
+                self._attr_native_value = response.json()["sum"]
+            elif self.entity_description.key == CONF_CURRENT_VALUES:
                 url_string = (
                     f"{_RESOURCE}getCurrentValuesSummary?token={self.app_token}"
                 )
                 response = requests.get(url_string, timeout=10)
                 for sensor in response.json():
                     if self.sid == sensor["sid"]:
                         measurement = next(iter(sensor["data"][0].values()))
-                        self._state = measurement
+                        self._attr_native_value = measurement
             else:
-                self._state = None
+                self._attr_native_value = None
```

Comment:
I think this can be removed. The config schema does not allow for extra keys so we would never reach this condition.